packages/task/src/browser/task-frontend-contribution.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable } from "inversify";
+import { ILogger } from '@theia/core/lib/common';
+import { QuickOpenTask } from './quick-open-task';
+import { MAIN_MENU_BAR, CommandContribution, Command, CommandRegistry, MenuContribution, MenuModelRegistry } from '@theia/core/lib/common';
+import { FrontendApplication } from '@theia/core/lib/browser';
+import { WidgetManager } from '@theia/core/lib/browser/widget-manager';
+import { WorkspaceService } from '@theia/workspace/lib/browser/workspace-service';
+
+export namespace TaskCommands {
+    // Task menu
+    export const TASK_MENU = [...MAIN_MENU_BAR, '3_task'];
+    export const TASK_MENU_RUN = [...TASK_MENU, '1_run'];
+    export const TASK_MENU_LABEL = 'Task';
+
+    // run task group
+    export const TASK = [...MAIN_MENU_BAR, '3_task'];
+    export const RUN_GROUP = [...TASK, '1_run'];
+
+    // run task command
+    export const TASK_RUN: Command = {
+        id: 'task:run',
+        label: 'Start New Task...'
+    };
+
+    export const TASK_ATTACH: Command = {
+        id: 'task:attach',
+        label: 'Running Tasks - List/Connect ...'
+    };
+}
+
+@injectable()
+export class TaskFrontendContribution implements CommandContribution, MenuContribution {
+
+    private isWorkSpaceSet = false;
+
+    constructor(
+        @inject(QuickOpenTask) protected readonly quickOpenTask: QuickOpenTask,
+        @inject(FrontendApplication) protected readonly app: FrontendApplication,
+        @inject(ILogger) protected readonly logger: ILogger,
+        @inject(WidgetManager) protected readonly widgetManager: WidgetManager,
+        @inject(WorkspaceService) protected readonly workspaceService: WorkspaceService
+    ) {
+        workspaceService.root.then(root => {
+            if (root !== undefined) {
+                this.isWorkSpaceSet = true;
+            }
+        });
+    }
+
+    registerCommands(registry: CommandRegistry): void {
+        registry.registerCommand(
+            TaskCommands.TASK_RUN,
+            {
+                isEnabled: () => this.isWorkSpaceSet,
```

Comment:
True. The idea was not to show the command as enabled unless the workspace is set. It avoids a potential issue with tasks: a task can have its "cwd" set/interpreted relative to the workspace. e.g. "... cwd: '$workspace/foo/' ". This will be an issue in the future if a task defined in user-storage (i.e. not in a specific WS) uses this feature, without having a WS set. 

That being said, ATM, we could remove this restriction, without bad consequences, I think.

Reply:
I think you should keep it considering you need the WS ATM ... and re-eavaluate when you're using user-storage...

Reply:
I don't think it is needed atm. So I'd rather remove it until it is needed. Maybe you want to disable unless there are tasks?

Reply:
Removed tie to workspace being set - now the command are always enabled.